src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int bucketIndex = calculateIndex(key);
+        Node<K, V> currentNode = table[bucketIndex];
 
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+
+        Node<K, V> newNode = new Node<>(key, value, table[bucketIndex]);
+        table[bucketIndex] = newNode;
+        size++;
+
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int bucketIndex = calculateIndex(key);
+        Node<K, V> currentNode = table[bucketIndex];
+
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int calculateIndex(K key) {
+        return (key == null) ? 0 : (key.hashCode() & 0x7FFFFFFF) % table.length;
+    }
+
+    private void resize() {
+        final Node<K, V>[] oldTable = table;
+        int newCapacity = table.length * 2;
+        table = new Node[newCapacity];
+        threshold = (int) (newCapacity * DEFAULT_LOAD_FACTOR);
+
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                int bucketIndex = calculateIndex(node.key);
+                Node<K, V> newNode = new Node<>(node.key, node.value, table[bucketIndex]);
+                table[bucketIndex] = newNode;
+                node = node.next;
+            }
+        }
```

Comment:
In the `resize` method, the `size` variable is not updated correctly. After rehashing the nodes into the new table, the `size` should reflect the actual number of elements in the hash map. You should iterate over the old table and update the `size` accordingly, or reset it and increment it as you add nodes to the new table.